src/scripts.js

```diff
@@ -107,28 +117,45 @@ function checkForHydrationInputs(event) {
 function addHydrationData() {
   const numOunces = parseFloat(hydrationOunces.value);
   const date = hydrationDate.value.split('-').join('/');
-
-  const userInput = { userID: userId, date, numOunces };
-
-  const postedData = postData('http://localhost:3001/api/v1/hydration', userInput);
-
-  postedData.then((data) => {
-    console.log(data);
-    hydrationResponse.innerText = 'Your hydration data was successfully uploaded!';
+  // check for date already existing...
+  if (checkIfDataExits(userId, date, allFetchedData[3])) {
+    hydrationResponse.innerText = `Data already exists for that date, please select a new date.`;
     hydrationResponse.classList.remove('hidden');
-    hydrationForm.classList.add('hidden');
     setTimeout(() => {
       domUpdates.hideResponse(hydrationResponse, hydrationForm);
-    }, 2500);
-  });
+    }, 2000);
+  } else {
+    const userInput = { userID: userId, date, numOunces };
+
+    const postedData = postData('http://localhost:3001/api/v1/hydration', userInput);
+
+    postedData.then((data) => {
+      console.log(data);
+      hydrationResponse.innerText = 'Your hydration data was successfully uploaded!';
+      hydrationResponse.classList.remove('hidden');
+      hydrationForm.classList.add('hidden');
+      setTimeout(() => {
+        domUpdates.hideResponse(hydrationResponse, hydrationForm);
+      }, 2500);
+    });
+
+    getAllData().then(data => {
+      const hydration = new Hydration(userId, data[3]);
+      hydrationDataChart.destroy();
+      // assigning the value of hydration chart a
+      // second time to be able to input data more than once
+      hydrationDataChart = domUpdates.renderHydration(hydration);
+    });
+  }
+}
 
-  getAllData().then(data => {
-    const hydration = new Hydration(userId, data[3]);
-    hydrationDataChart.destroy();
-    // assigning the value of hydration chart a 
-    // second time to be able to input data more than once
-    hydrationDataChart = domUpdates.renderHydration(hydration);
-  });
+function checkIfDataExits(id, date, dataArray) {
```

Comment:
Nice job -- this definitely seems like our best bet for stopping duplicate data.